src/main.rs

```diff
@@ -329,17 +335,22 @@ extern "C" fn kernel_init(_hart_id: u64, _fdt_addr: u64) {
     poweroff();
 }
 
+#[cfg(test)]
+#[no_mangle]
+extern "C" fn primary_launch() {}
```

Comment:
Would you consider using the naming `{primary,secondary}_stage{1,2}` as then you don't need to remember if init comes before launch?

Reply:
Not having strong opinion on  names, but believing that launch happens before init is a bit of a stretch.

Thought initially about stage0,1 but decided to do this essentially for fishing for names.

Reply:
current thinking (offline discussion with @rbradford) is to go towards `_init` and `_main`, almost crt-like. Open to change.